app-emulation/fs-uae/patches/fs_uae-2.8.3.patchset

```diff
@@ -0,0 +1,503 @@
+ 	libfsemu/src/init.c libfsemu/src/log.c libfsemu/src/malloc.c \
+ 	libfsemu/src/ml/clock.c libfsemu/src/ml/config.c \
+ 	libfsemu/src/ml/input.c libfsemu/src/ml/keyboard.c \
+-	libfsemu/src/ml/macosx.c libfsemu/src/ml/ml.c \
++	libfsemu/src/ml/macosx.c libfsemu/src/ml/haiku.c libfsemu/src/ml/ml.c \
+ 	libfsemu/src/ml/ml_internal.h libfsemu/src/ml/mouse.c \
+ 	libfsemu/src/ml/opengl.c libfsemu/src/ml/rawinput.c \
+ 	libfsemu/src/ml/render.c libfsemu/src/ml/sdl.c \
+@@ -326,7 +326,7 @@ am_libfsemu_a_OBJECTS = libfsemu/src/audio/audio-common.$(OBJEXT) \
+ 	libfsemu/src/ml/config.$(OBJEXT) \
+ 	libfsemu/src/ml/input.$(OBJEXT) \
+ 	libfsemu/src/ml/keyboard.$(OBJEXT) \
+-	libfsemu/src/ml/macosx.$(OBJEXT) libfsemu/src/ml/ml.$(OBJEXT) \
++	libfsemu/src/ml/macosx.$(OBJEXT) libfsemu/src/ml/haiku.$(OBJEXT) libfsemu/src/ml/ml.$(OBJEXT) \
+ 	libfsemu/src/ml/mouse.$(OBJEXT) \
+ 	libfsemu/src/ml/opengl.$(OBJEXT) \
+ 	libfsemu/src/ml/rawinput.$(OBJEXT) \
+@@ -1950,7 +1950,7 @@ libfsemu_a_SOURCES = libfsemu/include/fs/base.h \
+ 	libfsemu/src/init.c libfsemu/src/log.c libfsemu/src/malloc.c \
+ 	libfsemu/src/ml/clock.c libfsemu/src/ml/config.c \
+ 	libfsemu/src/ml/input.c libfsemu/src/ml/keyboard.c \
+-	libfsemu/src/ml/macosx.c libfsemu/src/ml/ml.c \
++	libfsemu/src/ml/macosx.c libfsemu/src/ml/haiku.c libfsemu/src/ml/ml.c \
+ 	libfsemu/src/ml/ml_internal.h libfsemu/src/ml/mouse.c \
+ 	libfsemu/src/ml/opengl.c libfsemu/src/ml/rawinput.c \
+ 	libfsemu/src/ml/render.c libfsemu/src/ml/sdl.c \
+diff --git a/configure b/configure
+index 1cea6de..440d824 100755
+--- a/configure
++++ b/configure
+@@ -8076,6 +8076,8 @@ case $host_os in #(
+      ;; #(
+   darwin*) :
+      ;; #(
++  haiku*) :
++     ;; #(
+   *) :
+ 
+     X11_LIBS="-lX11"
+@@ -8412,6 +8414,14 @@ esac
+     OS_CPPFLAGS="-DDRAGONFLYBSD"
+     LOADABLE_MODULE_EXT=".so"
+  ;; #(
++  haiku*) :
++
++    OS_NAME="haiku"
++    OS_NAME_2="Haiku"
++    OS_NAME_3="Haiku"
++    OS_CPPFLAGS="-DHAIKU"
++    LOADABLE_MODULE_EXT=".so"
++ ;; #(
+   *) :
+ 
+     OS_NAME="unknown"
+diff --git a/configure.ac b/configure.ac
+index 168533b..c8c4a80 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -270,6 +270,7 @@ AC_DEFINE([USE_SDL_INPUT], [1], [Define to 1 to use SDL for input])
+ AS_CASE([$host_os],
+ [mingw*], [],
+ [darwin*], [],
++[haiku*], [],
+ [
+     X11_LIBS="-lX11"
+     AC_DEFINE([USE_X11], [1], [Define to 1 to use X11])
+@@ -352,6 +353,13 @@ AS_CASE([$host_os],
+     OS_CPPFLAGS="-DDRAGONFLYBSD"
+     LOADABLE_MODULE_EXT=".so"
+ ],
++[haiku*], [
++    OS_NAME="haiku"
++    OS_NAME_2="Haiku"
++    OS_NAME_3="Haiku"
++    OS_CPPFLAGS="-DHAIKU"
++    LOADABLE_MODULE_EXT=".so"
++],
+ [
+     OS_NAME="unknown"
+     OS_NAME_2="Unknown OS"
+diff --git a/libfsemu/include/fs/endian.h b/libfsemu/include/fs/endian.h
+index 01ee0f9..5bbcd9f 100644
+--- a/libfsemu/include/fs/endian.h
++++ b/libfsemu/include/fs/endian.h
+@@ -7,6 +7,14 @@
+ 
+ #include <stdint.h>
+ 
++#ifdef __HAIKU__
```

Comment:
OK, this can now be done thanks to @jessicah with changes so that -D_BSD_SOURCE allows including <endian.h> with the endian conversion macros.
I will update the patch shortly.